### PR TITLE
Iot docs : add iot projects examples

### PR DESCRIPTION
**Pull request by jbtrystram**
### Type of change

- Documentation

### Description

Add some documentation around the iotproject.

### Checklist

- [x] Update documentation

### CODE REVIEW

**Comment by jbtrystram** [comment]
@dejanb I moved the config examples under "IoT for messaging tenants" as you advised

File: documentation/assemblies/assembly-iot-tenant-guide.adoc [hunk 1]
```diff
@@ -35,3 +35,10 @@ include::../modules/proc-iot-sending-events-mqtt.adoc[leveloffset=+2]
 
 include::assembly-iot-configuring-sigfox.adoc[leveloffset=+1]
 
+== IoT projects configuration examples
```
Comment (jenmalloy) [inline documentation/assemblies/assembly-iot-tenant-guide.adoc]:
```suggestion
== IoT project configuration examples
```

File: documentation/modules/ref-iot-project-external-example.adoc [hunk 2]
```diff
@@ -0,0 +1,13 @@
+// Module included in the following assemblies:
+//
+// assembly-iot-service-admin-guide.adoc
+
+[id='iot-project-external-example']
+= Using an external messaging infrastructure
+
+This IoT configuration example shows how an external messaging infrastructure can be provided.
```
Comment (jenmalloy) [inline documentation/modules/ref-iot-project-external-example.adoc]:
```suggestion
This IoT configuration example shows how an external messaging infrastructure can be configured.
```

File: documentation/modules/ref-iot-project-managed-example.adoc [hunk 3]
```diff
@@ -0,0 +1,19 @@
+// Module included in the following assemblies:
+//
+// assembly-iot-service-admin-guide.adoc
+
+[id='iot-project-managed-example']
+= Using a managed messaging infrastructure
+
+This IoT project configuration example rely on {productName} to manage the messaging infrastructure used by iot traffic.
```
Comment (jenmalloy) [inline documentation/modules/ref-iot-project-managed-example.adoc]:
```suggestion
This IoT project configuration example relies on {ProductName} to manage the messaging infrastructure used by IoT traffic.
```

File: documentation/modules/ref-iot-project-managed-example.adoc [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+Here we use {productName} standard address space and address plans.
```
Comment (jenmalloy) [inline documentation/modules/ref-iot-project-managed-example.adoc]:
```suggestion
The {ProductName} `standard` address space and address plans are used.
```

File: documentation/modules/ref-iot-project-managed-example.adoc [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+[source,yaml,options="nowrap"]
+----
+include::../common/iot-project-managed-example.yaml[]
+----
+<1> The `managedStrategy` value refers to an messaging infrastructure managed by {productName}.
```
Comment (jenmalloy) [inline documentation/modules/ref-iot-project-managed-example.adoc]:
```suggestion
<1> The `managedStrategy` value refers to a messaging infrastructure managed by {ProductName}.
```

File: documentation/modules/ref-iot-project-managed-example.adoc [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+<2> The address space plan, defining resource usage of the address space. Each iot tenant get his own address space.
```
Comment (jenmalloy) [inline documentation/modules/ref-iot-project-managed-example.adoc]:
```suggestion
<2> Specifies the address space plan, defining the resource usage of the address space. Each IoT tenant must have its own address space.
```

File: documentation/modules/ref-iot-project-managed-example.adoc [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+<3> Each address must be declared with an valid address plan.
```
Comment (jenmalloy) [inline documentation/modules/ref-iot-project-managed-example.adoc]:
```suggestion
<3> Each address must be associated with a valid address plan.
```

File: documentation/modules/ref-iot-project-managed-example.adoc [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+<4> The type of the address space. Default is `standard`. See link:{BookUrlBase}{BaseProductVersion}{BookNameUrl}#con-address-space-messaging[address space documentation] for more details.
```
Comment (jenmalloy) [inline documentation/modules/ref-iot-project-managed-example.adoc]:
```suggestion
<4> Specifies the type of the address space. The default value is `standard`. For more information see the link:{BookUrlBase}{BaseProductVersion}{BookNameUrl}#con-address-space-messaging[address space documentation].
```

File: documentation/modules/ref-iot-project-managed-example.adoc [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+<5> The type of the address. See link:{BookUrlBase}{BaseProductVersion}{BookNameUrl}#con-address-space-messaging[address space documentation] for more details.
```
Comment (jenmalloy) [inline documentation/modules/ref-iot-project-managed-example.adoc]:
```suggestion
<5> Specifies the type of address. For more information see link:{BookUrlBase}{BaseProductVersion}{BookNameUrl}#con-address-space-messaging[address space documentation].
```

**Comment by jbtrystram** [comment]
@dejanb like this ? I created a new assembly called `iot for project owners" that is called in `assembly iot guide`

**Comment by jbtrystram** [comment]
Should we remove the whole `assembly-iot-tenant-guide.adoc` if it's not used ?

**Comment by dejanb** [comment]
> Should we remove the whole assembly-iot-tenant-guide.adoc if it's not used ?

@jenmalloy I'm not sure if you need this assembly for anything else. If not we can remove it.

**Pull request merged**